src/LogicItLab/Salesforce/MapperBundle/WsdlValidator.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace LogicItLab\Salesforce\MapperBundle;
+
+use LogicItLab\Salesforce\MapperBundle\Annotation\AnnotationReader;
+use LogicItLab\Salesforce\MapperBundle\Exception\FileHandlerNotInitializedException;
+use Symfony\Component\Finder\Finder;
+
+class WsdlValidator
+{
+    /** @var AnnotationReader */
+    private $annotationReader;
+
+    /** @var string */
+    private $root;
+
+    /** @var string */
+    private $path;
+
+    /** @var string */
+    private $wsdlPath;
+
+    /**
+     * @param AnnotationReader $annotationReader
+     * @param string $root
+     * @param string $wsdlPath
+     * @codeCoverageIgnore
+     */
+    public function __construct(AnnotationReader $annotationReader, string $root, string $path, string $wsdlPath)
+    {
+        $this->annotationReader = $annotationReader;
+        $this->root = $root;
+        $this->path = $path;
+        $this->wsdlPath = $wsdlPath;
+    }
+
+    /**
+     * @return array
+     * @throws FileHandlerNotInitializedException
+     * @throws \ReflectionException
+     */
+    public function retrieveMissingFields(): array
+    {
+        $missingFields = [];
+
+        $fileHandler = fopen($this->wsdlPath, "r");
+        $this->initializeFileHandler($fileHandler);
+
+        foreach ($this->getAllClassAnnotations() as $annotation) {
+            $objectName = $annotation['object']->name;
+            $fieldNames = $this->getFieldNames($annotation['fields']);
+
+            if (!$objectName || !$fieldNames) {
+                continue;
+            }
+
+            if ($this->hasObject($fileHandler, $objectName) === false) {
+                $missingFields[$objectName] [] = "entire object";
+                continue;
+            }
+
+            foreach ($fieldNames as $fieldName) {
+                if ($this->hasField($fileHandler, $fieldName) === false) {
+                    $missingFields[$objectName] [] = $fieldName;
+                }
+            }
+        }
+
+        return $missingFields;
+    }
+
+    private function getAllClassAnnotations(): array
+    {
+        $annotations = array_map(function ($className) {
+            return $this->annotationReader->getSalesforceProperties($className);
+        }, $this->getAllClassNames());
+
+        usort($annotations, function ($a, $b) {
+            return strcmp($a['object']->name, $b['object']->name);
+        });
+
+        return $annotations;
+    }
+
+    private function getAllClassNames(): array
+    {
+        $filenames = [];
+        $finderFiles = Finder::create()->files()->in($this->root.$this->path)->name('*.php');
```

Comment:
I'm not sure what $finderFiles are wouldn't it make more sense to name them e.g. $foundFiles or $salesforceClasses or $salesforceModels?